### PR TITLE
Refactor pgp_key_t methods for noexcept and fix few coverity issues.

### PR DESCRIPTION
**Pull request by ni4**
...also return key material from pgp_key_t by pointer instead of references.

### CODE REVIEW

File: src/lib/pgp-key.cpp [hunk 1]
```diff
@@ -1596,7 +1590,9 @@ pgp_key_t::lock()
         return true;
     }
 
-    material().clear_secret();
+    if (material()) {
```
Comment (desvxx) [inline src/lib/pgp-key.cpp:1593]: maybe it will be good to add assert not null here

File: src/lib/pgp-key.cpp [hunk 2]
```diff
@@ -1687,7 +1683,9 @@ pgp_key_t::unprotect(const pgp_password_provider_t &password_provider,
     }
     pkt_ = std::move(*decrypted_seckey);
     /* current logic is that unprotected key should be additionally unlocked */
-    material().clear_secret();
+    if (material()) {
```
Comment (desvxx) [inline src/lib/pgp-key.cpp:1686]: maybe it will be good to add assert not null here

**Review by desvxx — APPROVED** [review]

File: src/lib/pgp-key.cpp (hunk 1 above)
Comment (ni4) [reply src/lib/pgp-key.cpp:1594]: Thanks, addressed this and the next comments.

File: .github/workflows/windows-native.yml
Comment (maxirmx) [inline .github/workflows/windows-native.yml:1]: I believe the root cause is here: #2150

**Review by maxirmx — APPROVED** [review]

**Review by antonsviridenko — APPROVED** [review]

**Comment by ni4** [comment]
Having approvals and green CI - let's merge it. Thanks all!

**Pull request merged**